### PR TITLE
Leader board Implementation

### PR DESCRIPTION
**Pull request by aleena-joseph-86**
https://github.com/user-attachments/assets/d828ea5f-d2e2-415c-a165-8c6152bac6d6

### CODE REVIEW

File: src/app/game/question-display/page.tsx [hunk 1]
```diff
@@ -1,30 +1,51 @@
...
 }: {
+  timer: number;
   game: Game;
   callback: (correct: boolean) => void;
 }) {
   let currentQuestion = game.questions[game.gameLevel - 1];
   const [selectedAnswer, setSelectedAnswer] = useState<number | null>(null);
   const [correctAnswer, setCorrectAnswer] = useState<number | null>(null);
 
+  const router = useRouter();
```
Comment (rmadhuram) [inline src/app/game/question-display/page.tsx:19]: I think the best way to handle this is outside this component.

File: src/app/home/page.tsx [hunk 2]
```diff
@@ -65,13 +44,17 @@ function Leaderboard({ title, leaderboard }: LeaderboardProps) {
...
 
 export default function Home() {
-  const router = useRouter()
-  const [name, setName] = useState('');
-  const [type, setType] = useState('kid');
+  const router = useRouter();
+  const [name, setName] = useState("");
+  const [type, setType] = useState("");
+  const [leaderboard, setLeaderBoard] = useState<{
+    kids: LeaderboardProps;
```
Comment (rmadhuram) [inline src/app/home/page.tsx]: Can we use the LeaderBoards model defined in model.ts?

File: src/app/api/leaderboard/route.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+export async function POST(request: Request) {
+  const { name, score, time, level } = await request.json();
+
+  const leaderboard = fetchLeaderBoard();
+
+  if (!(level in leaderboard)) {
+    return NextResponse.json({ error: "Invalid level" }, { status: 400 });
+  }
+
+  const updatedLeaderboard = {
```
Comment (rmadhuram) [inline src/app/api/leaderboard/route.ts:25]: Please move this and all logic to client/game-service.ts. 
It is better if route just calls functions from the services, and have no business logic.

File: src/app/game/question-display/page.tsx (hunk 1 above)
Comment (aleena-joseph-86) [reply src/app/game/question-display/page.tsx:19]: Ok Sir

File: src/app/home/page.tsx (hunk 2 above)
Comment (aleena-joseph-86) [reply src/app/home/page.tsx]: Sir, In the leaderBoards model defined in the models doesnt have title Sir. Can I add it ?

File: src/app/api/leaderboard/route.ts (hunk 3 above)
Comment (aleena-joseph-86) [reply src/app/api/leaderboard/route.ts:25]: Ok Sir

**Comment by rmadhuram** [comment]
nice job @aleena-joseph-86 👍

**Pull request merged**